src/contributor-guide/pull-request-tests.md

```diff
@@ -0,0 +1,131 @@
+
+Below there's information about all kinds of tests available currently.
+
+### Semantic Version Checker
+
+Semantic Version Checker is a test that compares the changed code against the mainline and detects levels of code changes. It shows a list of changes grouped by the class (level) of changes (MAJOR, MINOR, PATCH) with explanation of the change (Class was removed, [public] Method implementation changed, [private] Property has been removed etc.) The allowed level is PATCH.
+Please see the [Backward Compatibility Development Guide](https://devdocs.magento.com/guides/v2.3/contributor-guide/backward-compatible-development/index.html) for more information
+
+### Functional Tests
+Functional Tests are PHPUnit-based tests that check Magento functional behavior from the perspective of a user. They press buttons and links in browser, fill and submit forms, look and check what is displayed in the browser's window like real users.
+
+### Sample Data Tests
+Sample Data Tests are functional tests that run on Magento with Sample Data included. Currently they include two kinds of tests:
+* Sample Data functional tests which check Magento functional behavior with Sample Data installed
+* Installation Functional tests which check Magento installation flow with Sample Data installed
+
+### Unit Tests
+Unit tests are PHPUnit-based (for PHP) and Jasmine-based (for JS) tests which check the behavior of single methods of Magento classes. The purpose is to validate that each unit of Magento performs as designed.
+
+### Integration Tests
+Integration Tests are PHPUnit-based tests that check the behaviour of group of Magento components. They include three types of tests:
+* Standard Integration tests which check Magento behavior on installed instance.
+* Setup Integration tests which check Magento behavior during the installation process.
+* Integration Integrity tests which check the consistency of installed Magento instance (configuration files, DB schemes etc.)
+
+### WebAPI Tests
+WebAPI Tests are functional tests that check the behavior of Magento Web API endpoints. They send requests to endpoints of installed Magento instance and and then check the response received and the state of the instance. Currently they include three kinds of tests:
+* SOAP Web API tests
+* REST Web API tests
+* GraphQL WEB API tests
+
+### Static Tests
+Static Tests are set of different checks that analyze each single file of Magento code base separately. Most of them analyze the whole Magento code base whereas some run only against the change set. They include:
+* Static Integration tests (PHPUnit-based) which check the consistency of Magento instance without installation (configuration files, DB schemes etc.)
```

Comment:
Also for some kind of tests there were added failure reports. I don't have exact list of test builds. Please talk to Ann Beeskau

Reply:
@abeeskau can you please help to update this documentation content?

Reply:
@slavvka see the comment above. The details have been removed from the PR